samples/template-operator/operator/controllers/sample_controller.go

```diff
@@ -82,3 +86,26 @@ func transform(_ context.Context, _ types.BaseCustomObject, manifestResources *t
 	}
 	return nil
 }
+
+// ManifestResolver represents the chart information for the passed Sample resource.
+type ManifestResolver struct{}
+
+// Get returns the chart information to be processed.
+func (m *ManifestResolver) Get(obj types.BaseCustomObject) (types.InstallationSpec, error) {
```

Comment:
I think we should implement a ManifestResolver for channels. I will prepare that separately so this PR can go in regardless.